### PR TITLE
Added support of toggling substring search.

### PR DESCRIPTION
**Pull request by sealionkat**
UIE-527

#### What has changed in the code ####

- toggling between old and new filtering
- checking whether substring search is enabled in schema
- filter mode toggle in navbar menu
- saving data about substring search in sessionStorage and Redux store
- added searching by ID

#### Reasons for making this change ####
    
Possibility between old and new filtering.

### Checklist

* [x] **I'm adding or updating code**
  - [x] I've added and/or updated tests
  - [ ] I've updated docs if needed
* [ ] **I'm adding a new feature**
  - [ ] I've added unit test for feature
  - [ ] I've ran lint

### CODE REVIEW

File: src/config/ConfigActions.js [hunk 1]
```diff
@@ -87,3 +96,17 @@ export function fetchAppVersion() {
     });
   };
 }
+
+export function setSubstringSearchEnabled(enabled) {
```
Comment (stepek) [inline src/config/ConfigActions.js]: `isEnabled`

File: src/TableView/TableView.jsx [hunk 2]
```diff
@@ -78,15 +82,26 @@ export const getTableView = (schema, Table = TableComponent, isChildView = false
 
     componentDidMount() {
       if (!isChildView) {
-        const query = queryParse(this.props.location.search);
+        const {substringSearchEnabled, metadataSubstringSearchEnabled} = this.props;
+        const queryFormat = substringSearchEnabled ||
+          metadataSubstringSearchEnabled === false ?
```
Comment (michal-damiecki) [inline src/TableView/TableView.jsx]: Please use consistent syntax when `substringSearchEnabled` and `metadataSubstringSearchEnabled` are used with logical operators. They are both boolean values so it's not necessary to compare them with boolean value.
```suggestion
          !metadataSubstringSearchEnabled ?
```

File: src/TableView/TableView.jsx [hunk 3]
```diff
@@ -257,16 +287,33 @@ export const getTableView = (schema, Table = TableComponent, isChildView = false
     };
 
     handleFilterData = property => {
-      const searchFields = property === undefined ? undefined : Object.keys(property);
+      const {substringSearchEnabled, metadataSubstringSearchEnabled} = this.props;
+      const searchFields = substringSearchEnabled === true &&
```
Comment (michal-damiecki) [inline src/TableView/TableView.jsx]: Please use consistent syntax when `substringSearchEnabled` and `metadataSubstringSearchEnabled` are used with logical operators. They are both boolean values so it's not necessary to compare them with boolean value.
```suggestion
      const searchFields = substringSearchEnabled &&
```

File: src/TableView/TableView.jsx [hunk 4]
```diff
@@ -257,16 +287,33 @@ export const getTableView = (schema, Table = TableComponent, isChildView = false
...
+        property !== undefined ?
+        Object.keys(property) :
+        undefined;
+      const queryFormat = substringSearchEnabled ||
+        metadataSubstringSearchEnabled === false ?
+        undefined :
+        {arrayFormat: 'bracket'};
+      const filters = (substringSearchEnabled === false || metadataSubstringSearchEnabled === false) &&
```
Comment (michal-damiecki) [inline src/TableView/TableView.jsx]: Please use consistent syntax when `substringSearchEnabled` and `metadataSubstringSearchEnabled` are used with logical operators. They are both boolean values so it's not necessary to compare them with boolean value.
```suggestion
      const filters = (!substringSearchEnabled || !metadataSubstringSearchEnabled) &&
```

File: src/TableView/TableView.jsx [hunk 5]
```diff
@@ -398,7 +447,9 @@ export const getTableView = (schema, Table = TableComponent, isChildView = false
             onChange: this.handleFilterData,
             by: filterBy,
             value: filterValue,
-            onlyStringTypes: true
+            onlyStringTypes: true,
+            includeRelations: !substringSearchEnabled || metadataSubstringSearchEnabled === false,
```
Comment (michal-damiecki) [inline src/TableView/TableView.jsx]: Please use consistent syntax when `substringSearchEnabled` and `metadataSubstringSearchEnabled` are used with logical operators. They are both boolean values so it's not necessary to compare them with boolean value.
```suggestion
            includeRelations: !substringSearchEnabled || !metadataSubstringSearchEnabled,
```

**Pull request merged**